promql/parser/printer.go

```diff
@@ -147,22 +147,37 @@ func (node *BinaryExpr) ShortString() string {
 
 func (node *BinaryExpr) getMatchingStr() string {
 	matching := ""
+	var b bytes.Buffer
 	vm := node.VectorMatching
 	if vm != nil {
 		if len(vm.MatchingLabels) > 0 || vm.On || vm.Card == CardManyToOne || vm.Card == CardOneToMany {
 			vmTag := "ignoring"
 			if vm.On {
 				vmTag = "on"
 			}
-			matching = fmt.Sprintf(" %s (%s)", vmTag, strings.Join(vm.MatchingLabels, ", "))
+			// Format labels with proper UTF-8 quoting using writeLabels()
+			b.Reset()
+			b.WriteString(" ")
+			b.WriteString(vmTag)
+			b.WriteString(" (")
```

Comment:
I think `b.WriteString(" " + vmTag + " (")` should be sufficient here, let's not blow it up more than we need to :)

Similar below for `" group_" + vmCard + " ("`.